src/Nethermind/Nethermind.Serialization.Rlp/ReceiptArrayStorageDecoder.cs

```diff
@@ -19,12 +19,7 @@ public class ReceiptArrayStorageDecoder : IRlpStreamDecoder<TxReceipt[]>
     private static readonly IRlpValueDecoder<TxReceipt> CompactValueDecoder = Rlp.GetValueDecoder<TxReceipt>(RlpDecoderKey.Storage);
 
     public const int CompactEncoding = 127;
-    private readonly bool _useCompactEncoding = true;
-
-    public ReceiptArrayStorageDecoder(bool compactEncoding = true)
-    {
-        _useCompactEncoding = compactEncoding;
-    }
+    private readonly bool _useCompactEncoding = compactEncoding;
```

Comment:
this field is probably not needed now.

Reply:
If you mean we can drop non compact encoding, there a small concern: IReceiptConfig mentions that usage of compact encoding may effect RPC performance. Does it mean non compact receipts may still have users? If no - let's remove old encoding for sure

Reply:
I mean you can use default constructor parameter `compactEncoding` and don't need the field.

Reply:
Oh indeed